ctapipe/calib/camera/calibrator.py

```diff
@@ -25,27 +30,15 @@ class CameraCalibrator(Component):
     .. code-block:: python
 
         aliases = Dict(dict(
-        ped='CameraR1CalibratorFactory.pedestal_path',
-        tf='CameraR1CalibratorFactory.tf_path',
-        pe='CameraR1CalibratorFactory.adc2pe_path',
-        extractor='ChargeExtractorFactory.extractor',
-        extractor_t0='ChargeExtractorFactory.t0',
-        window_width='ChargeExtractorFactory.window_width',
-        window_shift='ChargeExtractorFactory.window_shift',
-        sig_amp_cut_HG='ChargeExtractorFactory.sig_amp_cut_HG',
-        sig_amp_cut_LG='ChargeExtractorFactory.sig_amp_cut_LG',
-        lwt='ChargeExtractorFactory.lwt',
```

Comment:
Could you restore all of these aliases?

I guess it would need to refer to the actual class where these aliases are created? For example in place of `ped='CameraR1CalibratorFactory.pedestal_path',` one would do `ped='TargetIOR1Calibrator.pedestal_path',`

I suppose some traitlets may need to be moved to their parent classes to avoid the need to repeat this for every R1Calibrator that uses a `pedestal_path`. This is for a different PR.

Reply:
I think we should write down what we want here: I would say, having the ability to set configuration parameters from the base class and any extra ones in the sub-class is a nice thing, and the developer should not have to repeat them in each case. But maybe this points more to a problem with the configuration system.  I'm still in favor of eventually dropping the implementation of traitlets.config.Application and rolling our own simpler one (still using traitlets, since I do like them).  E.g. just defining that any traitlet in a class is a configurable parameter.

Reply:
I think these aliases cannot be restored with this current approach, since at Tool creation time it is not known which of the ChargeExtractors will be selected by the user.

This is a major difference between the current version in the master and this proposal here: A user can only configure things by naming them explicitly. I believe this is an advantage, since errors are more visible in the command line call.

As an example, this is a valid (but probably wrong) invocation using the current master:

    ctapipe-event-viewer \
    --ChargeExtractorFactory.t0=10  \
    --ChargeExtractorFactory.sig_amp_cut_LG=5.3

No existing ChargeExtractor has both `t0` and `sig_amp_cut_LG`.

Contrasting this branch here, forces the user to write this:
    ctapipe-event-viewer \
    --SimpleIntegrator.t0=10  \
    --GlobalPeakIntegrator.sig_amp_cut_LG=5.3

Which gives a user another moment to pause and say ... hey, I might have made a mistake here.

Reply:
From your comment @dneise, it does sound like it can be restored, it just has to be more verbose than the current implementation:

```python
        ped='TargetIOR1Calibrator.pedestal_path',	
        tf='TargetIOR1Calibrator.tf_path',	
        pe='TargetIOR1Calibrator.adc2pe_path',	
        simpleintegrator_t0='SimpleIntegrator.t0',	
        window_width='WindowIntegrator.window_width',	
        window_shift='WindowIntegrator.window_shift',	
        sig_amp_cut_HG='PeakFindingIntegrator.sig_amp_cut_HG',	
        sig_amp_cut_LG='PeakFindingIntegrator.sig_amp_cut_LG',	
        lwt='NeighbourPeakIntegrator.lwt',
```
then one can do `ctapipe-event-viewer --window_width 20`
Could you see if that works?

Reply:
> E.g. just defining that any traitlet in a class is a configurable parameter.

I believe this is possible, as in @dneise comment:
```
ctapipe-event-viewer \
--ChargeExtractorFactory.t0=10  \
--ChargeExtractorFactory.sig_amp_cut_LG=5.3
```

Its only when you wish to set an alias does one need to write out the extra stuff - which I think is important as it avoids confusion between identical traitlets. Does one also need to have the class included in the `classes` List to be able to do the `--SimpleIntegrator.t0=10`?

Reply:
> I think we should write down what we want here: I would say, having the ability to set configuration parameters from the base class and any extra ones in the sub-class is a nice thing

This is what I think could be convenient, is this you meant too @kosack?

The following should be valid to set a Charge Extractor window width:

```python
ctapipe-event-viewer --WindowIntegrator.window_width=10 #Works in current version
ctapipe-event-viewer --PeakFindingIntegrator.window_width=10 #Does not yet work (I think)
ctapipe-event-viewer --LocalPeakIntegrator.window_width=10 #Does not yet work (I think)
```

As all these classes are subclasses of `WindowIntegrator`, where the `window_width` traitlet is defined.

However I don't think the following should be possible as it might cause problems, as different subclasses of this class may have their own version of the traitlet which operates differently. I think this behaviour partially exists with the current Factory implementation, which @dneise is trying to avoid:

```python
ctapipe-event-viewer --ChargeExtractor.window_width=10
```

Reply:
So interestingly, even this works to change the `window_width` in both this PR and the master:

```
ctapipe-event-viewer --Component.window_width=20
```
You can tell by scrolling down on the viewer to see the value its been set to. It does result in a bunch of warnings for all the Components that don't have a window_width traitlet. But it seems that as long as the class is a subclass of what you specify on the command line, it will try to set the traitlet value.

Reply:
>  ... it does sound like it can be restored ...

Ah .. I didn't get that. In that case I will happily do that ... soon.

Reply:
I restored these aliases in the last commits (travis is running the tests currently) so I assume this conversation could be closed.